lib/promscrape/scraper.go

```diff
@@ -256,7 +256,11 @@ func (scfg *scrapeConfig) run(globalStopCh <-chan struct{}) {
 		sws := scfg.getScrapeWork(cfg, swsPrev)
 		sg.update(sws)
 		swsPrev = sws
-		scfg.discoveryDuration.UpdateDuration(startTime)
+		if sg.scrapersStarted.Get() > 0 {
```

Comment:
My thinking was that scenario when vmagent was configured with SD and then suddenly was updated to not use SD - is very rare. It is unlikely something like this would happen frequently.